### PR TITLE
add loading indicator

### PR DESCRIPTION
**Pull request by li-qiang**

### CODE REVIEW

File: addon/controllers/grouped-row-array.js [hunk 1]
```diff
@@ -170,6 +174,10 @@ export default RowArrayController.extend({
   },
 
   depthFirstTravers: function(content, callback, level) {
```
Comment (jefflage) [inline addon/controllers/grouped-row-array.js]: typo on the name

File: addon/controllers/grouped-row-array.js [hunk 2]
```diff
@@ -71,7 +71,8 @@ export default RowArrayController.extend({
     var self = this;
     controllersMap.forEach(function(value) {
```
Comment (jefflage) [inline addon/controllers/grouped-row-array.js]: what is `_controllersMap`?

File: addon/controllers/grouped-row-array.js [hunk 3]
```diff
@@ -134,6 +135,9 @@ export default RowArrayController.extend({
     var childrenRows = this.get('_childrenRows');
 
     while (childrenRows.has(theObject)) {
+      if(theObject.hasLoadedChildren === false){
+        return {object: this._loadingObjectFor(theObject), level: theLevel, parent: theObject};
```
Comment (jefflage) [inline addon/controllers/grouped-row-array.js]: what does this small inline hash represent?  does this imply some expectation of all of data you pass to the table?  should this be some kind of model object?

File: addon/controllers/grouped-row-array.js (hunk 1 above)
Comment (jefflage) [inline addon/controllers/grouped-row-array.js]: could there be a better name than `callback`?  maybe something to explain WHY you're calling out to an external?

File: addon/controllers/grouped-row-array.js (hunk 3 above)
Comment (cyjia) [inline addon/controllers/grouped-row-array.js]: Not needed, it's part of group data provider.

File: addon/controllers/grouped-row-array.js (hunk 2 above)
Comment (cyjia) [inline addon/controllers/grouped-row-array.js]: a map between row content and row controller

File: addon/controllers/grouped-row-array.js (hunk 1 above)
Comment (cyjia) [inline addon/controllers/grouped-row-array.js]: fixed in #35

**Comment by cyjia** [comment]
merged in #35

**Pull request closed without merging**